api/oc_rep_encode.c

```diff
@@ -181,100 +222,117 @@ oc_rep_encoder_set_type_by_accept(oc_content_format_t accept)
   return false;
 }
 
-oc_content_format_t
+int
```

Comment:
Why int ? If you want to signal error change definintion to:
`bool oc_rep_encoder_get_content_format(ct* oc_content_format_t)`